plugin/meshnet.go

```diff
@@ -84,32 +84,43 @@ func loadConf(bytes []byte) (*netConf, *current.Result, error) {
 }
 
 // getVxlanSource uses netlink to get the iface reliably given an IP address.
-func getVxlanSource(nodeIP string) (string, string, error) {
-	if nodeIP == "" {
-		return "", "", fmt.Errorf("meshnetd provided no HOST_IP address: %s", nodeIP)
-	}
-	nIP := net.ParseIP(nodeIP)
-	if nIP == nil {
-		return "", "", fmt.Errorf("parsing failed for meshnetd provided no HOST_IP address: %s", nodeIP)
-	}
-	ifaces, _ := net.Interfaces()
-	for _, i := range ifaces {
-		addrs, _ := i.Addrs()
-		for _, a := range addrs {
-			var ip net.IP
-			switch v := a.(type) {
-			case *net.IPNet:
-				ip = v.IP
-			case *net.IPAddr:
-				ip = v.IP
-			}
-			if nIP.Equal(ip) {
-				log.Infof("Found iface %s for address %s", i.Name, nodeIP)
-				return nodeIP, i.Name, nil
-			}
-		}
-	}
-	return "", "", fmt.Errorf("no iface found for address %s", nodeIP)
+// when IP and Interface both are present then Interface is going to take preference
+// nodeIntf is specified by the user and it's not auto discovered. The user has to be careful that the peer is reachable though this interface otherwise, VxLAN may not work.
+// daemonset.yaml meshnet container env required for host_intf override
+//          env:
+//            - name: HOST_INTF
+//   			value: breth2
```

Comment:
can you fix up  the formatting here?

Reply:
Sure @networkop i have pushed the fix for this.